src/controllers/UserController.js

```diff
@@ -139,17 +141,26 @@ class UserController {
     // check if password looks alike
     if (req.body.password === req.body.confirm_password) {
       try {
+        // check whether token is valid
+        const verifyToken = await findToken(req.params.token);
+        if (!verifyToken.valid === true) {
+          util.setError(400, 'Token has been used');
```

Comment:
I don't think we need to explicitly tell the user about token expiration. Consider using a user friendly message e.g. you have already reset your password, request another reset if you don't remember your password

Reply:
Done